backend/ligand_loader.py

```
@@ -0,0 +1,975 @@
+
+            print(f"\n🔍 Loading ligands for target {target_id} ({gene_symbol or uniprot_id})...")
+
+            # Determine interaction type from ligand_types (needed for all data sources)
+            ligand_types = target.get('ligand_types', [])
+            interaction_type = 'inhibitor'  # default
+            if ligand_types:
+                if isinstance(ligand_types, str):
+                    import json
+                    try:
+                        ligand_types = json.loads(ligand_types)
+                    except:
```

Ruff (0.15.2)

[error] Do not use bare `except`

(E722)